c10/core/impl/PyInterpreter.cpp

```diff
@@ -5,6 +5,19 @@
 namespace c10 {
 namespace impl {
 
+template<typename... Ts>
+static void noop_trace_cuda_fn(const PyInterpreter*, Ts...) {}
```

Comment:
Can't we technically apply this pattern everywhere?

Actually, I've been a bit irritated at the need to manually do function pointers and disarm them one by one, and I'm wondering if there's a way we can use plain old virtual methods to do this. 

```
// WARNING: This class has to be written very carefully, because it may be
// possible for a Tensor to have a reference an interpreter corresponding to
// a shared library that has ALREADY BEEN UNLOADED.  This makes blindly calling
// virtual methods very dangerous, because the vtable may be garbage at that
// point (on a good day, you might get "pure virtual method called").
//
// The idea to solve this problem is we always leak PyInterpreters (so they
// always stay live even after dlclose), and disarm the "virtual methods" by
// replacing them with function pointers that just no-op.  This can't be done
// with a traditional C++ vtable, so we have to roll our own.
```

I wonder if we can solve the problem that was specified here by doing a second layer of indirection. So the outer PyInterpreter is non-virtual but contains a pointer to an object with the virtual methods, and then unloading simply involves swapping out that pointer for a different one that is guaranteed to be in a stable shared library. The cost is an extra indirection to get to the actual function pointer, maybe that's a small price to pay for dramatically decreasing the boilerplate here (when this class was originally written, it was like... two function pointers. Now there's so many lol.)